core/src/avm2/array.rs

```diff
@@ -21,8 +245,10 @@ impl<'gc> ArrayStorage<'gc> {
     /// The length parameter indicates how big the array storage should start
     /// out as. All array storage consists of holes.
     pub fn new(length: usize) -> Self {
-        Self {
-            storage: vec![None; length],
+        if length > (1 << 28) {
```

Comment:
Can this `(1 << 28)` be moved to the top of the file, like `const MAX_DENSE_LENGTH`?